### DISCUSSION TITLE
[New] Respecter les bonnes pratiques de conception

### DISCUSSION THREAD

**Discussion by ddebarbieux** [cnumr/best-practices#107] [Bonnes pratiques] [answered]
Avant d'éco-concevoir, il faut concevoir et donc respecter cette phase :

1. Décrire l'architecture dans un document, les choix techniques sont débatus par l'équipe technique...
2. Mettre en place des outils de qualité logicielle

(liste à compléter si besoin)

**Comment by fzaninotto** [comment]
Quel impact sur les émissions de CO2, la consommation d'eau, l'extraction de métaux rares ?

**Reply by ddebarbieux** [reply]
J'ai déjà aidé avec une équipe qui appliquait mécaniquement les 115 bonnes pratiques. Mais dans l’équipe, personne ne pouvait me décrire l’architecture technique.

Je me suis dit qu’il n’avait rien compris à l’éco-conception.

**Comment by Trankilloman** [comment]
Plutôt d'accord avec ce point qui est cohérent avec ce que j'exprime ici https://github.com/cnumr/best-practices/discussions/88

**Comment by ACTLEM** [comment]
Plutôt que d'être une bonne pratique d'écoconception (en accord avec @fzaninotto que cela n'entre pas dans ce registre), et poursuivant l'idée de @Trankilloman, il faudrait une section dédiée (comme déjà proposé pour les outils) pour rappeler la démarche d'écoconception.
La qualité logicielle en est un prérequis, l'intelligence collective et l'argumentation des choix de conception (par ex un ADR) sont des aidants à la démarche, et les mentionner me plaît bien.

**Reply by devcrafting** [reply]
J'aime bien l'idée de le mentionner (car c'est en parti mon dada), mais je me demande pourquoi la qualité logicielle serait un pré-requis? Il y a même certaines pratiques qui sont limite à l'encontre de la qualité logicielle (même si plusieurs sont en passe d'être supprimées 😉).

Il serait intéressant de bien l'argumenter dans une section avant les BP. Idem pour intelligence collective et l'argumentation des choix de conception. Je suis convaincu mais il ne faut pas que ce soit trop péremptoire pour ne pas créer un biais dès le début au lecteur.

**Reply by ddebarbieux** [reply]
Malheureusement, beaucoup de lecteurs parcours les 115 bonnes pratiques et oublient de lire l’introduction. De même, lors d’un ‘audit’, seules les 115 bonnes pratiques sont vérifiées.

C’est pourquoi je propose d’un faire une nouvelle recommandation.

**Reply by florinesueur** [reply]
Dans ce sens, l'introduction n'est pas présente sur [https://collectif.greenit.fr/ecoconception-web/115-bonnes-pratiques-eco-conception_web.html](https://collectif.greenit.fr/ecoconception-web/115-bonnes-pratiques-eco-conception_web.html). 

Comme il est potentiellement possible d'avoir, une section mesure de "**Performance environnementale et amélioration continue**" proposait par @ACTLEM #36 , il est aussi possible d'avoir une section de "**Prérequis**" comprenant notamment :

- le cadrage (cahier des charges, qualité logicielle, contrat de service ...) de sa démarche #88 #117 
- l'intelligence collective avec toutes les parties prenantes (beaucoup pensent encore que l'écoconception est juste l'affaire des devs)
- le business model (est-il en accord avec une démarche d'écoconception ?) #59 
- étudier les solutions low tech en amont
- ...

**Reply by rlemaire** [reply]
Je suis d'accord. Beaucoup de gens se focalisent vite sur les pratiques en oubliant pourquoi on les liste.

Il faut bien insister sur **pourquoi** elles sont la, sur le fait que c'est une liste d'idées d'améliorations, sans obligations de les appliquer toutes, etc...

**Reply by ACTLEM** [reply]
> mais je me demande pourquoi la qualité logicielle serait un pré-requis? 

Pour répondre rapidement: Ici, je parle plus dans le sens testing, détection des anomalies et facilité de maintenance. Un service qui contient trop d'anomalies, est compliquée à maintenir dès le début va forcément avoir plus d'impacts (utilisé pour rien, effort supplémentaire à maintenir voir augmentation de l'infra pour palier une anomalie issue de non qualité...).

**Reply by ACTLEM** [reply]
Concernant la recommandation, j'aimerais avoir l'avis de @Trankilloman. Est-ce que cela doit rentrer dans le cadre d'une bonne pratique d'écoconception que de respecter l'état de l'art (car il s'agit ici d'un de peu cela)?
J'avoue que si j'écoconçois un produit physique (genre un smartphone), avoir la BP "Avoir les schémas détaillés technique" me semble naturel et n'est pas en soit de l'écoconception mais plus un prérequis.

**Reply by Trankilloman** [reply]
Cela peut être une bonne idée comme une très mauvaise idée. Si en effet l'état de l'art te dit "il faut avoir des schémas détaillés" alors ok car c'est assez général et toujours d'actualité. Si par contre l'état de l'art te dit "il faut avoir imprimé et envoyé 10 exemplaires (signés/contre signés et contre contre signés) de obligatoirement 10 schémas technique" alors c'est peut être pas (ou plus) nécessaire pour son système. Du coup, je dirais plutôt qu'il est nécessaire de suivre l'état de l'art (des pratiques industrielles, pas que de conception) EN les questionnants sur la pertinence technologique, géographique, temporelle... En fonction de son UF et périmètre de son système et de ses possibilités d'actions. Cette recommandation s'appliquant à l'utilisation même des BP du guide.

**Reply by Trankilloman** [reply]
Cela ferait d'ailleurs une superbe transition entre une introduction qui alerterait sur ce point et le début des BP ;-)

**Comment by ACTLEM** [comment, marked as answer]
https://github.com/cnumr/best-practices/issues/139

**Comment by Christophe-Cl** [comment]
OK pour cette BP générique de bon sens général. Cela peut inclure :
- une réflexion sur le réel besoin et la nécessité d'un service numérique (logiciel, outil, app, site, objet connecté, etc.) ou d'une fonctionnalité
- une réflexion sur la démarche prévue en terme de revues de conception ou de revues de code, avec impact environnemental inclus
- une réflexion sur la stratégie projet, notamment l'approche Green-UX qui a des répercutions sur tout le process, les collaborateurs et les utilisateurs finaux

Toutefois ATTENTION à la formulation afin d'éviter qu'une BP ne devienne "Respecter les BP ...". :)

**Reply by ACTLEM** [reply]
L'équipe projet a validé l'idée d'une page hors bonne pratique. En effet, comme tu le dis c'est plus une démarche qu'une BP opérationnelle en soit.
